src/ca_roots.jl

```diff
@@ -84,29 +84,34 @@ NetworkOptions could only find OpenSSL-specific TLS certificates which cannot be
 
 function system_ca_roots()
     lock(SYSTEM_CA_ROOTS_LOCK) do
-        isassigned(SYSTEM_CA_ROOTS) && return # from lock()
-        search_path = Sys.islinux() ? LINUX_CA_ROOTS :
-            Sys.isbsd() && !Sys.isapple() ? BSD_CA_ROOTS : String[]
-        openssl_only = false
-        for path in search_path
-            ispath(path) || continue
-            for line in eachline(path)
-                if line == BEGIN_CERT_REGULAR
-                    SYSTEM_CA_ROOTS[] = path
-                    return # from lock()
-                elseif line == BEGIN_CERT_OPENSSL
-                    openssl_only = true
+        roots = SYSTEM_CA_ROOTS[]
+        if roots === nothing
+            search_path = Sys.islinux() ? LINUX_CA_ROOTS :
+                Sys.isbsd() && !Sys.isapple() ? BSD_CA_ROOTS : String[]
+            openssl_only = false
+            for path in search_path
+                ispath(path) || continue
+                for line in eachline(path)
+                    if line == BEGIN_CERT_REGULAR
+                        roots = path
+                        openssl_only = false
```

Comment:
Why does this change more of the logic than just `SYSTEM_CA_ROOTS`? Was there another issue to fix here?

Reply:
I'll change it back, I just wanted it to be more similar to how it was changed in BUNDLED_KNOWN_HOSTS_FILE

Reply:
That's fine, I just wasn't sure about the openssl_only part, which looks like a logic change to me.